### PR TITLE
[k] nn category ops fix 

### PR DESCRIPTION
**Pull request by RobertLuobo**
Extract the nn (neural network) functional subset of the dev_ops commit: nn-related ops (attention, batch_norm, dropout, embedding, softmax, binary_cross_entropy_with_logits, nanmedian, selu, etc.), their registration entries in ops/__init__.py, and the nn smoke test (test_polar.py).

Also fix pre-commit lint issues in the extracted ops:
- add noqa: F401 for op names discovered via reflection
- rename ambiguous E741 variables (l, O) and drop dead F841 locals

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/runtime/backend/_kunlunxin/ops/attention.py [hunk 1]
```diff
@@ -1095,6 +1229,40 @@ def flash_attention_forward(
...
+        row = torch.arange(q_len, device=q.device)[:, None]
+        col = torch.arange(k_len, device=q.device)[None, :]
+        relative_pos = (row + k_len - q_len - col).abs()
+        scores = scores - alibi_slopes[..., None, None] * relative_pos
+    lse = torch.logsumexp(scores, dim=-1)
+    attn = torch.softmax(scores, dim=-1)
+    if dropout_p:
+        attn = torch.dropout(attn, dropout_p, True)
+    out = torch.matmul(attn, v).transpose(1, 2)
+    return out, lse, None, None, None
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/attention.py]: This path silently ignores `softcap` and `seqused_k`, and always returns `None` for the Philox state and debug mask. Please preserve these API semantics or explicitly reject unsupported options.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/nansum.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        out_dtype = dtype
+
+    if dim is None or dim == []:
+        res = _nansum_global(inp, out_dtype)
+        if dim == [] and keepdim:
+            res = res.reshape([1] * inp.ndim)
+        return res
+
+    dims = [dim] if isinstance(dim, int) else list(dim)
+    dims = [d % inp.ndim for d in dims]
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/nansum.py:329]: Please validate dimensions before normalization. Using d % inp.ndim silently turns out-of-range dimensions such as 3 or -4 for a 3-D tensor into valid axes, whereas ATen raises an error; duplicate dimensions should also be rejected.

Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/nansum.py:329]: DONE

File: src/flag_gems/runtime/backend/_kunlunxin/ops/attention.py (hunk 1 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/attention.py]: revert the attention.py change.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/selu.py [hunk 3]
```diff
@@ -39,11 +46,85 @@ def selu_func(x):
...
+    x_neg = tl.minimum(x_f32, 0.0)  # clamp exp arg to avoid overflow on x>0
+    y = _SCALE * tl.where(x_f32 > 0.0, x_f32, _ALPHA * (tl.exp(x_neg) - 1.0))
+
+    if NEED_MASK:
+        tl.store(O + offsets, y.to(x.dtype), mask=mask)
+    else:
+        tl.store(O + offsets, y.to(x.dtype))
+
+
+def _pick_tier(numel):
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/selu.py:83]: The same inline tiling selection appears in the other operator in this PR, so please move these tuning parameters into the vendor heuristics config.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/scaled_mm.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    elif out.shape != (M, N):
+        raise RuntimeError("Incompatible output shape")
+    if M == 0 or N == 0:
+        return out
+
+    scale_a, _ = _normalize_scale(scale_a, M, is_left_scale=True)
+    scale_b, _ = _normalize_scale(scale_b, N, is_left_scale=False)
+    bias = _normalize_bias(bias, N)
+
+    result = torch.mm(self.to(torch.float32), mat2.to(torch.float32))
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/scaled_mm.py]: Dont directly call torch.mm

File: src/flag_gems/runtime/backend/_kunlunxin/ops/fused_recurrent_gated_delta_rule_fwd.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            sequence if ssm_state_indices is None else ssm_state_indices[begin].item()
+        )
+        for value_head in range(value_heads):
+            query_head = value_head // (value_heads // heads)
+            state = source_state[initial_idx, value_head].float().clone()
+            for position in range(begin, end):
+                query = q[batch_idx, position, query_head].float()
+                key = k[batch_idx, position, query_head].float()
+                value = v[batch_idx, position, value_head].float()
+                if use_qk_l2norm_in_kernel:
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/fused_recurrent_gated_delta_rule_fwd.py]: Dont directly call torch api

File: src/flag_gems/runtime/backend/_kunlunxin/ops/rnn_relu.py [hunk 6]
```diff
@@ -62,32 +103,134 @@ def rnn_relu(
...
+                hh_t = (
+                    h.matmul(w_hh.t()) + b_hh
+                    if b_hh is not None
+                    else h.matmul(w_hh.t())
+                )
+                h = torch.relu(ih_t.to(torch.float32) + hh_t.to(torch.float32)).to(
+                    x.dtype
+                )
+                outputs.append(h)
+            output = torch.stack(outputs, 0)
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/rnn_relu.py]: Dont directly call torch api.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/smooth_l1_loss.py [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        if reduction == 0:
+            return torch.empty(shape, device=input.device, dtype=input.dtype)
+        if reduction == 1:
+            return torch.full((), float("nan"), device=input.device, dtype=input.dtype)
+        return torch.zeros((), device=input.device, dtype=input.dtype)
+
+    loss = _loss_values(input_expanded, target_expanded, beta)
+    if reduction == 0:
+        return loss
+    result = torch.sum(loss)
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/smooth_l1_loss.py]: Dont directly call torch api

File: src/flag_gems/runtime/backend/_kunlunxin/ops/selu.py (hunk 3 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/selu.py:83]: image
I think this tiling-size config is ok right here due to only 2 config.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/rnn_relu.py (hunk 6 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/rnn_relu.py]: DONE

File: src/flag_gems/runtime/backend/_kunlunxin/ops/smooth_l1_loss.py (hunk 7 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/smooth_l1_loss.py]: DONE

File: src/flag_gems/runtime/backend/_kunlunxin/ops/scaled_mm.py (hunk 4 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/scaled_mm.py]: DONE

File: src/flag_gems/runtime/backend/_kunlunxin/ops/fused_recurrent_gated_delta_rule_fwd.py (hunk 5 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/fused_recurrent_gated_delta_rule_fwd.py]: DONE

**Review by Caeruleann — DISMISSED** [review]

**Review by 103yiran — APPROVED** [review]
LGTM
